packages/webpack-cli/src/plugins/dotenv-webpack-plugin.js

```diff
@@ -0,0 +1,172 @@
+import dotenv from "dotenv-defaults";
+
+const interpolate = (env, vars) => {
+  const matches = env.match(/\$([a-zA-Z0-9_]+)|\${([a-zA-Z0-9_]+)}/g) || [];
+  matches.forEach((match) => {
+    env = env.replace(match, interpolate(vars[match.replace(/\$|{|}/g, "")] || "", vars));
+  });
+  return env;
+};
+
+const isMainThreadElectron = (target) => target.startsWith("electron") && target.endsWith("main");
+
+class Dotenv {
+  constructor(config = {}) {
+    this.config = Object.assign(
+      {},
+      {
+        path: "./.env",
+        prefixes: ["process.env.", "import.meta.env."],
+      },
+      config,
+    );
+    this.cache = {};
```

Comment:
That is a potential memory leaking cache, please use webpack cache, you can find an example in https://github.com/webpack-contrib/copy-webpack-plugin/blob/master/src/index.js or WeakMap